src/components/Leaderboard.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { FirebaseClassContext } from './App';
+import './styles/Leaderboard.scss';
+
+interface Score {
+  name?: string;
+  email?: string;
+  level?: string;
+  score?: number;
+}
+function Leaderboard(): JSX.Element {
+  const context = useContext(FirebaseClassContext);
+
+  const [scores, setScores] = useState<Array<Score>>([]);
+  useEffect(() => {
+    context.firebase.getTopScores().then((s: Array<Score>) => {
+      setScores(
+        s
+          .sort(function (a, b) {
+            return (b.score || 0) - (a.score || 0);
+          })
+          .slice(0, 10),
+      );
```

Comment:
```suggestion
      setScores(
        s.sort((a, b) => (b.score ?? 0) - (a.score ?? 0)).slice(0, 10),
      );
```
use the nullish coalesce operator

Reply:
also can we have the `_Firebase` class handle this? 

```ts
public getTopScores(num_scores?: number): UserInfo[]
```

Reply:
yeah good idea tyy